src/g_umapinfo.c

```diff
@@ -285,7 +270,7 @@ static void ParseStandardProperty(scanner_t *s, mapentry_t *mape)
         {
             if (!strcasecmp(SC_GetString(s), "clear"))
             {
-                ReplaceString(&mape->label, "-");
+                mape->flags |= MapInfo_LabelClear;
```

Comment:
I have never understood why we would have `char[]` arrays in the umapinfo structs and check their first char for being `-` and similar. Why don't we just have `char *` pointers and it pointing to `NULL` will mean "cleared" and it pointing to some allocated string would mean this string.

Reply:
In the case of the `label` field, we need a separate "clear" state to remove the default label (e.g. "MAP01:"), so I added the `MapInfo_LabelClear` flag. We need a lot of flags, so it's also cumbersome.
`char[]` arrays indicate lump names wich should be 8 charaters max.

Reply:
> What I'm asking myself is, why did we set `mape->label[0] = '-'` instead of just letting `mape->label = NULL`? This would be a much clearer distinction between an empty string (`""`) and a cleared string (`NULL`).

Because we need 3 states in UMAPINFO:
1. `label = "Level 1"`, map title: "Level 1: Entryway"
2. No label field, `label = NULL` - use default, map title: "MAP01: Entryway".
3. `label = clear` - remove the default label, map title: "Entryway".

In this PR I have replaced `'-'` strings with flags.

Reply:
I see. We could replace the third state by a pointer to an empty string, but I'll leave this up to you of course.

Reply:
I think pointer to empty string is not much different then `"-"` string. I added flags because it's more readable, but that's a matter of taste.